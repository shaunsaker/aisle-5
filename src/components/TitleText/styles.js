import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

const styles = StyleSheet.create({
  text: {
    ...styleConstants.fonts.types.title,
  },
});

export default styles;
