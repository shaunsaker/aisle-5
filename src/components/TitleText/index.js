import React from 'react';
import PropTypes from 'prop-types';
import { Text } from 'react-native';

import styleConstants from '../../styleConstants';

import styles from './styles';

const propTypes = {
  text: PropTypes.string,
  color: PropTypes.string,
};

const defaultProps = {
  color: styleConstants.colors.white,
};

const TitleText = ({ text, color }) => {
  return <Text style={[styles.text, { color }]}>{text}</Text>;
};

TitleText.propTypes = propTypes;
TitleText.defaultProps = defaultProps;

export default TitleText;
