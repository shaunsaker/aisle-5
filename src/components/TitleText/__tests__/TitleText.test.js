import React from 'react';
import renderer from 'react-test-renderer';

import TitleText from '..';

describe('TitleText', () => {
  describe('renders', () => {
    it('renders with minimum required props', () => {
      const component = renderer.create(<TitleText />);

      expect(component).toMatchSnapshot();
    });
  });
});
