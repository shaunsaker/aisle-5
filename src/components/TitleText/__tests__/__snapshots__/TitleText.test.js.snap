// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleText renders renders with minimum required props 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": "#212121",
        "fontFamily": "ArchivoBlack-Regular",
        "fontSize": 24,
      },
      Object {
        "color": "#EDE8CA",
      },
    ]
  }
/>
`;
